addons/l10n_it_edi/models/res_partner.py

```diff
@@ -148,3 +146,49 @@ def validate_codice_fiscale(self):
         for record in self:
             if record.l10n_it_codice_fiscale and (not codicefiscale.is_valid(record.l10n_it_codice_fiscale) and not iva.is_valid(record.l10n_it_codice_fiscale)):
                 raise UserError(_("Invalid Codice Fiscale '%s': should be like 'MRTMTT91D08F205J' for physical person and '12345670546' for businesses.", record.l10n_it_codice_fiscale))
+
+    def _l10n_it_edi_export_check(self, partner_kind="other"):
+        def validate(key, list_of_fields, message, message_company, kinds=None):
+            if kinds and partner_kind not in kinds:
+                return {}
+            invalid_partners = self.env['res.partner']
+            for _fields in list_of_fields:
+                invalid_partners |= self.filtered(lambda record: not any(record[field] for field in _fields))
+            if len(invalid_partners) == 1:
+                views = [(False, 'form')]
+            else:
+                list_view = (self.env.ref('l10n_it_edi.res_partner_tree_l10n_it', raise_if_not_found=False))
+                views = [(list_view.id if list_view else False, 'list'), (False, 'form')]
+            return {key: {
+                'message': message,
+                'message_company': message_company,
+                'action_text': _("View Partner(s)"),
+                'action': self.env['ir.ui.view'].get_records_action(invalid_partners, name=_("Check Partner(s)"), views=views),
+            }} if invalid_partners else {}
+
+        return {
+            **validate(
+                key='partner_vat_missing',
+                list_of_fields=[('vat',)],
+                message=_("Partner(s) should have a VAT number."),
+                message_company=_("Company/ies should have a VAT number."),
+                kinds=['representative']),
+            **validate(
+                key='partner_vat_codice_fiscale_missing',
+                list_of_fields=[('vat', 'l10n_it_codice_fiscale')],
+                message=_("Partner(s) should have a VAT number or Codice Fiscale."),
+                message_company=_("Company/ies should have a VAT number or Codice Fiscale."),
+                kinds=['simplified', 'other']),
+            **validate(
+                key='partner_country_missing',
+                list_of_fields=[('country_id',)],
+                message=_("Partner(s) should have a Country when used for simplified invoices."),
+                message_company=_("Company/ies should have a Country when used for simplified invoices."),
+                kinds=['simplified']),
+            **validate(
+                key='partner_address_missing',
+                list_of_fields=[('street', 'street2'), ('zip',), ('city',), ('country_id',)],
```

Comment:
nitpick: why the trailing commas? it won't reduce any future diff. same above
```suggestion
                list_of_fields=[('street', 'street2'), ('zip'), ('city'), ('country_id')],
```

Reply:
my bad, I just realised it needs to be tuples. sorry

Reply:
These are tuples, not "trailing commas".
![image](https://github.com/odoo/odoo/assets/1665365/74c92709-2265-471f-b7d5-1f530aaefab7)